### ISSUE TITLE
Resizable column view

### ISSUE THREAD

**Issue by jainal09** [files-community/Files#6534]
### What's the Problem?

For files having long names. In column view it only covers some part of name. Having a feature to resize the column by clicking is absolutely necessary and needed!
(SEE SCREEN SHOT)
![image](https://user-images.githubusercontent.com/34179361/137611071-cc7152f2-e75a-4934-91fc-c22b8be5a411.png)

### Solution/Idea

Having a feature to resize the column by clicking.

### Alternatives

Not using column mode. But I really like it.

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish W | Must |
| This proposal will allow end users to accomplish X | Should |
| This proposal will allow developers to accomplish Y | Could |
| This proposal will allow end users to accomplish Z | Won't |

### Files Version

Version: 2.0.13.0 OS Version: 10.0.22478.1012

### Windows Version

Windows 11 22478.1012

**Comment by jainal09** [comment]
@yaichenbaum at least not a scroll bar. But just a  feature to select and drag the line

**Comment by sergiubacioiu** [comment]
An option in settings menu to resize all the columns would be great too. I mostly use a maximum of 2/3 subfolders so most of the time 30% of my screen is empty.

**Comment by yaira2** [comment]
Are the custom column sizes supposed to be saved for whenever the column is opened again?

**Comment by andrefromsbg** [comment]
hello, is there anything happening with this task? files is really a great file browser for windows. but the fact that you can't change the column width keeps me, and probably many others who are used to this rudimentary function of apple finder, from using files as a permanent alternative...

**Comment by yaira2** [comment]
> Are the custom column sizes supposed to be saved for whenever the directory is opened again?

@jainal09

**Comment by michael-oz-smith** [comment]
Saving column sizes would be helpful, but simply having the ability to resize columns is an ASAP necessity.

Users coming over from Macs are very accustomed to having this capability since its been an OS X feature since launch.

**Comment by yaira2** [comment]
Does MacOS save the sizes?

**Comment by Lukiluc29** [comment]
> Does MacOS save the sizes?

I believe and I even think that if this function is not on MacOS it would be better to put it like that we don't need to readjust the columns all the time to see the names of the documents

**Comment by jainal09** [comment]
> 

No macos doesnt save the resized column sizes

**Comment by Lukiluc29** [comment]
Yes but it's still better to include this option if possible and if it's not too long to code

**Comment by redhotideas** [comment]
@yaira2 yes, the request is the same, but I think that the columns must maintain a minimum size, the current one is fine.
Then enlarge to the right where necessary and be able to scale it back to its original size, but not smaller.

The autofit function would be useful by default to bring columns with already hidden names to the correct size to read everything

**Comment by andrefromsbg** [comment]
Hey people, this issue is now open for almost 2 years - is there a chance that something will happen here and this feature will be implemented?

**Comment by yaira2** [comment]
@andrefromsbg it's on the ToDo list but we're currently blocked by the lack of a proper DataGrid control in WinUI https://github.com/microsoft/microsoft-ui-xaml/issues/1500.

**Comment by andrefromsbg** [comment]
@yaira2 thank you for the quick reply and as always a big thank you to everyone involved in the development of this great application.

**Comment by mdtauk** [comment]
Maybe there should be a 3.5 max set on the columns, so 3 full columns and one half.

As the window gets narrower, this then goes down to 2.5 columns.

I guess there can be a 1.5 lower limit.

The column widths would resize with the window.  We should also consider the option of having two lines of text for the items.

**Comment by PatrickECleary** [comment]
Just curious... is there any movement on this FR? 

I, too, really like the column layout. But it's not usable when filenames are long.

Cocoatech nailed it in their [Path Finder](https://www.cocoatech.io/) software (Finder alternative on Mac). Their column view has an auto-size feature. _Very_ handy.

**Comment by Josh65-2201** [comment]
No one has started this yet. When someone start work there will be a PR linked showing it.

**Comment by panbeep** [comment]
Any updates on this? I still can't resize columns in latest 3.0.15 version

**Comment by yaira2** [comment]
@panbeep it hasn't been added yet

**Comment by panbeep** [comment]
@yaira2 is it possible to set/add a custom width for all columns? It might be temporary solution until resize is added.

**Comment by yaira2** [comment]
@panbeep it's about the same amount of work to implement this properly.

**Comment by polyclick** [comment]
Files has been a godsend for me after switching from 25 years of using MacOS. Column-view is such a speed boost in productivity.  But the inability to resize columns baffles me, can we please have this feature?

**Comment by SoTHISIsFitness** [comment]
This feature request is marked 'ready to build' - has anyone taken the lead on this / is there any update?

**Comment by 0x5bfa** [comment]
Someone did but failed to get it right.
For the devs, our plans around here is not to use BladeView anymore and use a Grid control with Grid.Children for each column view. This is what we did for shell panes and you can see the code here:

https://github.com/files-community/Files/blob/0debeee681f3443808ca49d47dc3e829c8a5e795/src/Files.App/Views/ShellPanesPage.xaml#L27

**Comment by mdtauk** [comment]
Ideally we would match the column width to the longest text string item, as this is how macOS Finder handles their column layout

**Comment by yair100** [comment]
Support for resizing columns will be included in the next release.

> Ideally we would match the column width to the longest text string item, as this is how macOS Finder handles their column layout

We're looking into this 👍

**Comment by Josh65-2201** [comment]
@jainal09 Is this still relevant to you? In v4+ columns now size based on the biggest name of the items

**Comment by ccrooks562** [comment]
You all did a FANTASTIC job!!! Thank you!!!!!

**Comment by jainal09** [comment]
@Josh65-2201 thanks for checking with me , tbh I stopped caring about it long time ago man. I opened this issue in 2021 and up till then, I switched to mac and linux, completed a new masters degree and switched countries. So, you can go ahead and close it if its fixed. Idc about this anymore


I just didn't close it myself for folks out here yearning to get this feature.

**Comment by yair100** [comment]
We're going to leave this open to continue tracking the request to sizes across sessions.
